### DISCUSSION TITLE
grpc 和 quic 对比普通的tcp+tls 到底有什么优势呀？

### DISCUSSION THREAD

**Discussion by lhbdhr** [v2fly/v2ray-core#2460] [Q&A]
grpc 和 quic 对比普通的tcp+tls 到底有什么优势呀？有没有详细介绍的文章或视频呀？
包含优势和劣势的？我搜了半天也没搞明白。据说grpc容易被探测？
很困惑，不知道是不是要用。

**Comment by mydogshitgold** [comment]
IMO it would be a challenge to [passively analyse](https://scholar.google.com/scholar?hl=en&as_sdt=0%2C5&q=passive+traffic+analysis+tls&btnG=) anything in a TLS stream other than the obvious TLS fingerprints / packet length.

gRPC has native support for multiplexing, which is better than the mux.cool, and is also supported by major CDNs like Cloudflare as well as web servers such as NGINX.  QUIC offers better congestion control, but no CDN provider supports QUIC-to-origin.

There are some example configurations on [v2fly/v2ray-examples](https://github.com/v2fly/v2ray-examples)

**Reply by lhbdhr** [reply]
thanks, i will check that.
